### PR TITLE
Pylint fixes for tcms.testplans.views

### PR DESCRIPTION
**Pull request by ivo0126**

### CODE REVIEW

File: tcms/testplans/views.py [hunk 1]
```diff
@@ -536,89 +543,12 @@ def clone(request, template_name='plan/clone.html'):
         return HttpResponseRedirect(request.META.get('HTTP_REFERER', '/'))
 
     # Clone the plan if the form is submitted
-    if request.method == "POST":
+    if request.method == "POST":  # pylint: disable=no-else-return
```
Comment (atodorov) [inline tcms/testplans/views.py]: Don't disable the check, remove the `else` part of this statement.

File: tcms/testplans/forms.py [hunk 2]
```diff
@@ -130,8 +130,9 @@ def clean(self, data, initial=None):
 
 class PlanModelForm(forms.ModelForm):
     class Meta:
-        model = TestPlan
-        exclude = ('author', )
+        def __init__(self):
```
[the remark this answers is not in the document]
Comment (ivo0126) [reply tcms/testplans/forms.py]: It throws a `modelform-uses-exclude` and [here](https://github.com/PyCQA/pylint-django/issues/168) it says that it might become a security issue if it isn't fixed.
But also i don't think i fully understand what `Use explicit fields instead of exclude in ModelForm` means.

File: tcms/testplans/forms.py [hunk 3]
```diff
@@ -176,7 +177,7 @@ class BasePlanForm(forms.Form):
         required=False
     )
 
-    def clean_parent(self):
+    def clean_parent(self):  # pylint: disable=inconsistent-return-statements
```
[the remark this answers is not in the document]
Comment (ivo0126) [reply tcms/testplans/forms.py]: Isn't pylint's problem that the method is explicitly told to return something only when a certain condition is met and that it wants me to explicitly return something else if the condition is not met?
From what i can see, the method returns either a testplan, `None` or raises an error and the only inconsistency in return statements is that it hasn't been explicitly told to return `None` if all else fails.

File: tcms/testplans/forms.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (atodorov) [reply tcms/testplans/forms.py]: Read this https://docs.djangoproject.com/en/2.1/topics/forms/modelforms/ and see the examples. Then ping me if they are not clear.

File: tcms/testplans/forms.py (hunk 3 above)
[the remark this answers is not in the document]
Comment (atodorov) [reply tcms/testplans/forms.py]: Hmm, that sounds like a bug with pylint itself. I will examine this checker and get back on this comment.

[the remark this answers is not in the document]
Comment (atodorov) [reply tcms/testplans/forms.py]: So according to the help documentation:

```
pylint --help-msg inconsistent-return-statements
:inconsistent-return-statements (R1710): *Either all return statements in a function should return an expression, or none of them should.*
  According to PEP8, if any return statement returns an expression, any return
  statements where no value is returned should explicitly state this as return
  None, and an explicit return statement should be present at the end of the
  function (if reachable) This message belongs to the refactoring checker.
```


so to silence this error you have to put a `return None` at the end.

File: tcms/testplans/views.py (hunk 1 above)
Comment (ivo0126) [reply tcms/testplans/views.py]: If i move the return outside, the below code will become unreachable.

File: tcms/testplans/admin.py [hunk 4]
```diff
@@ -21,7 +21,7 @@ class TestPlanAdmin(ReadOnlyHistoryAdmin):
     def add_view(self, request, form_url='', extra_context=None):
         return HttpResponseRedirect(reverse('admin:testplans_testplan_changelist'))
 
-    def change_view(self, request, object_id, extra_context=None):
+    def change_view(self, request, object_id, extra_context=None):  # pylint: disable=arguments-differ
```
[the remark this answers is not in the document]
Comment (atodorov) [reply tcms/testplans/admin.py]: Fix this so I can cherry-pick

File: tcms/testplans/forms.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (atodorov) [reply tcms/testplans/forms.py]: Fix this commit so I can cherry-pick

File: tcms/testplans/views.py [hunk 5]
```diff
@@ -361,69 +361,76 @@ def get(request, plan_id, slug=None, template_name='plan/get.html'):
 
 @require_http_methods(['GET', 'POST'])
 @permission_required('testruns.change_testrun')
-def choose_run(request, plan_id):
+def choose_run(request, plan_id):  # pylint: disable=inconsistent-return-statements
```
Comment (atodorov) [inline tcms/testplans/views.py]: This requires a bit of non-straight forward thinking. The reason for this warning is that you have 2 if statement which could return a response and then you have the last line of the function body, which by default is equivalent to "return None". The easiest way to fix that is to simply add this explicit return at the end.

However you have a decorator which indicates that only GET and POST methods are accepted and the second if statement is really the ELSE part for the first if statement.

On top of that the two branches of this function don't have anything in common so you can split it out into 2 separate functions that will each handle the respective http request method.

File: tcms/testplans/views.py (hunk 1 above)
Comment (atodorov) [reply tcms/testplans/views.py]: The reason for this is because by doing all the changes below you have modified the behaviour of the function. Now you have:
```

if request.method == 'POST':
    .....
    return ...
else:
   ....
   return ....

....
return
```

before that the else block didn't have a return statement. As with the comment above it may be easier to split this lare function into 2 smaller ones, each of which will handle the GET and POST requests separately; make sure these work and then fix whatever issues are reported by pylint.

File: tcms/testplans/views.py [hunk 6]
```diff
@@ -759,7 +780,8 @@ def post(self, request, plan_id):
 class DeleteCasesView(View):
     """Delete selected cases from plan"""
 
-    def post(self, request, plan_id):
+    @staticmethod
+    def post(request, plan_id):
```
Comment (atodorov) [inline tcms/testplans/views.py]: Class based view have the signature as shown above, however they may not use the self variable. You should revert the changes and open an issue against pylint-django with this code snippet so we can teach pylint-django to ignore these error messages for class based views.
